src/__tests__/TimelineComponent.test.tsx

```diff
@@ -198,6 +199,204 @@ describe('<TimelineComponent /> R2 Tests\n', () => {
+  const prettyHTML = (htmlString: string) => {
+    const lines: Array<string> = htmlString.split(/(?=<)/g);
+    let tempString: string = '';
+    for (let i: number = 0; i < lines.length; i++) {
+      tempString += (lines[i] + '\n');
+    }
+    return tempString;
+  };
+
+  let wrapper: any;
+
+  beforeEach(() => {
+    // need to do this so that d3 works properly with jest/enzyme
+    wrapper = mount(
+        <TimelineComponent
+          data={data}
+        />);
+    document.body.innerHTML = wrapper.html();
+    wrapper = mount(
+        <TimelineComponent
+          data={data}
+        />);
+    // initialize & draw the timeline (updates html)
+    wrapper.instance().resetTimeline();
+
+    // Check that rendered TimelineComponent HTML matches snapshot
+    // This doesn't include the html created by d3
+    expect(wrapper.debug()).toMatchSnapshot();
+
+    // check d3 svgs match the snapshot
+    const svgTarget: any = document.getElementById('svgtarget');
+    let svgHTML: string = '';
+    if (svgTarget) {
+      svgHTML = prettyHTML(svgTarget.innerHTML);
+    }
+    // Check that svg created by d3 matches snapshot
+    expect(svgHTML).toMatchSnapshot();
+  });
+  // zoom in events for keydown & keyup
+  const zoomInEventDown = new KeyboardEvent('keydown', {'key': '+'});
+  const zoomInEventUp = new KeyboardEvent('keyup', {'key': '+'});
+  // zoom out events for keydown & keyup
+  const zoomOutEventDown = new KeyboardEvent('keydown', {'key': '-'});
+  const zoomOutEventUp = new KeyboardEvent('keyup', {'key': '-'});
+
+
+  it('T3.3 Snapshot Test\n', () => {
+    expect(wrapper.instance().getScale()).toBe(1.0);
+
+    // check scale for zoomInEventDown and zoomInEventUp separately
+    document.body.dispatchEvent(zoomInEventDown);
+    document.body.dispatchEvent(zoomInEventUp);
+    const tempScaleEventDown = wrapper.instance().getScale();
+    expect(tempScaleEventDown).toBeGreaterThan(1.0);
+
+    // find d3 svg element
+    const svgTarget: any = document.getElementById('svgtarget');
+    let svgHTML: string = '';
+    if (svgTarget) {
+      svgHTML = prettyHTML(svgTarget.innerHTML);
+    }
+
+    // Check that svg created by d3 matches snapshot
+    expect(svgHTML).toMatchSnapshot();
+  });
+
+  it('T3.4 Snapshot test', ()=>{
+    // zoom in first so could test zoom backout
+    document.body.dispatchEvent(zoomInEventDown);
+    document.body.dispatchEvent(zoomInEventUp);
+    expect(wrapper.instance().getScale()).toBeGreaterThan(1.0);
+    // zoom back out
+    document.body.dispatchEvent(zoomOutEventDown);
+    document.body.dispatchEvent(zoomOutEventUp);
+    expect(wrapper.instance().getScale()).toBe(1.0);
+
+    // find d3 svg element
+    const svgTarget: any = document.getElementById('svgtarget');
+    let svgHTML: string = '';
+    if (svgTarget) {
+      svgHTML = prettyHTML(svgTarget.innerHTML);
+    }
+
+    // Check that svg created by d3 matches snapshot
+    expect(svgHTML).toMatchSnapshot();
```

Comment:
This doesn't even really test anything. It just makes sure it renders properly and doesn't actually add any value to the project since we test this already. The zooming thing is tested elsewhere as well.

You should be snapshot testing the zoom in, then zoom out and see that it returned to the original state.